src/Perl6/Optimizer.nqp

```diff
@@ -1899,7 +1899,7 @@ class Perl6::Optimizer {
                         $op.shift; $op.shift; # name, package (both pre-resolved now)
                         $op.unshift($inv);
                         $op.unshift($call);
-                        $op.op('call');
+                        $op.op('callstatic');
                         $op.name(NQPMu);
```

Comment:
I'm decidedly confused. :-) Here's the code that compiles `callstatic` differently from `call`:

https://github.com/perl6/nqp/blob/b6d98eac3f0dff1dbd4df252351efa35ab700f7e/src/vm/moar/QAST/QASTOperationsMAST.nqp#L1444

Notice how it is contained within a `if $op.name { ... }` block. All it's doing is instructing the VM that it can cache the lexical lookup, because it will always return the exact same code object (which is why we use it for things from the setting scope and `UNIT` scope - because they can never be closure-cloned).

Given that, I don't currently see how this change could affect the generated code, and so lead to a performance difference. I wonder what I'm missing...

Reply:
(Also, to be clear, I didn't yet attempt to replicate the performance difference locally.)